src/main/java/broker/LogSegment.java

```diff
@@ -0,0 +1,77 @@
+package broker;
+
+import org.tinylog.Logger;
+
+import java.io.File;
+import java.io.IOException;
+
+public class LogSegment {
+    private final int partitionNumber;
+    private File logFile; // stores the records
+    private boolean isActive; // if isActive is true, the segment is mutable
+    private long segmentThresholdInBytes =  1_048_576; // log segment cannot exceed this size threshold (default: 1 MB)
+    private long currentSizeInBytes;
+    private int startOffset; // for entire segment
+    private int endOffset; // for entire segment
+
+    public LogSegment(int partitionNumber, int startOffset) {
+        this.partitionNumber = partitionNumber;
+        this.startOffset = startOffset;
+
+        try {
+            // ex: partition1_000000.log, actual kafka names log files purely by byte offset like 000000123401.log
+            this.logFile = new File(String.format("./data/partition%d_%05d.log", this.partitionNumber, this.startOffset));
+            if (this.logFile.createNewFile()) {
+                Logger.info(String.format("File created: %s%n", this.logFile.getPath()));
+            } else {
+                Logger.warn(String.format("File %s already exists", this.logFile.getPath()));
+            }
+        } catch (IOException e) {
```

Comment:
You're good to merge the PR, but I noticed that since this is catching the io error, it's possible that the file could be null in the case of an error. Wanted to get your thoughts on possibly rethrowing the error